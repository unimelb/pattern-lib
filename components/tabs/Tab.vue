<template>
  <transition
    name="fade"
    mode="out-in">
    <div
      v-show="isActive"
      :id="`${namespace}-panel-${index + 1}`"
      class="tabs__panel max"
      role="tabpanel">
      <slot />
    </div>
  </transition>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: '',
    },
    active: {
      type: Boolean,
      default: false,
    },
  },
  data: () => ({
    isActive: false,
    namespace: '',
    index: 0,
  }),
  created() {
    this.isActive = this.active;
  },
};
</script>

<style>
.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.25s ease 0.125s;
}

.fade-enter,
.fade-leave-to {
  visibility: hidden;
  position: absolute;
  opacity: 0;
}
</style>
