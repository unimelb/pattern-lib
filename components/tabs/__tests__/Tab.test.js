import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import Tab from '../Tab.vue';

expect.extend(toHaveNoViolations);

describe('Tab', () => {
  it('should match snapshot', () => {
    const result = shallow(Tab).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(Tab).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
