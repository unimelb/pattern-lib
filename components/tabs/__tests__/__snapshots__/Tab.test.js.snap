// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should match snapshot 1`] = `
<div
  class="tabs__panel max"
  id="-panel-1"
  mode="out-in"
  name="fade"
  role="tabpanel"
  style="display: none;"
/>
`;
